### PR TITLE
fix(api): Add logging to rate limiting

### PR DESCRIPTION
**Pull request by schew2381**

### CODE REVIEW

File: src/sentry/middleware/ratelimit.py [hunk 1]
```diff
@@ -91,6 +92,9 @@ def process_view(self, request: Request, view_func, view_args, view_kwargs) -> R
                     request.will_be_rate_limited = True
                     enforce_rate_limit = getattr(view_class, "enforce_rate_limit", False)
                     if enforce_rate_limit:
+                        logger.exception(
+                            f"Rate limit exceeded for key `{request.rate_limit_key}` hitting {request.build_absolute_uri()}"
```
Comment (schew2381) [inline src/sentry/middleware/ratelimit.py:96]: Are these the correct things to be logging?

Comment (AniketDas-Tekky) [reply src/sentry/middleware/ratelimit.py:96]: I think this is gonna create an issue every time someone exceed the rate limit. let's just do `logger.info`

File: src/sentry/middleware/ratelimit.py [hunk 2]
```diff
@@ -91,6 +92,9 @@ def process_view(self, request: Request, view_func, view_args, view_kwargs) -> R
                     request.will_be_rate_limited = True
                     enforce_rate_limit = getattr(view_class, "enforce_rate_limit", False)
                     if enforce_rate_limit:
+                        logger.info(
```
Comment (AniketDas-Tekky) [inline src/sentry/middleware/ratelimit.py:95]: This won't show up properly in google logging. You'll need to do something like
```
logger.info("sentry.api.ratelimit.exceeded", extra={"key": request.rate_limit_key})
```

see:   https://github.com/getsentry/sentry/blob/3cfa5c961f86ae2054bc34388896f88ce4b0208b/src/sentry/integrations/vsts/webhooks.py#L44 as an example

File: src/sentry/middleware/ratelimit.py [hunk 3]
```diff
@@ -91,6 +92,13 @@ def process_view(self, request: Request, view_func, view_args, view_kwargs) -> R
...
+                            "sentry.api.rate-limit.exceeded",
+                            extra={
+                                "key": request.rate_limit_key,
+                                "url": request.build_absolute_uri(),
```
Comment (AniketDas-Tekky) [inline src/sentry/middleware/ratelimit.py:99]: Can you also capture the maximum size of the bucket?

**Review by AniketDas-Tekky — APPROVED** [review]

**Pull request merged**